low-level/sensors/impl/generic/node_data.py

```diff
@@ -157,11 +157,14 @@ def _get_host_update_data(self):
         # Calculate the current number of running processes at this moment
         total_running_proc = 0
         for proc in psutil.process_iter():
-            pinfo = proc.as_dict(attrs=['status'])
-            if pinfo['status'] not in (psutil.STATUS_ZOMBIE, psutil.STATUS_DEAD,
-                                       psutil.STATUS_STOPPED, psutil.STATUS_IDLE,
-                                       psutil.STATUS_SLEEPING):
-                total_running_proc += 1
+            try:
+                pinfo = proc.as_dict(attrs=['status'])
+                if pinfo['status'] not in (psutil.STATUS_ZOMBIE, psutil.STATUS_DEAD,
+                                        psutil.STATUS_STOPPED, psutil.STATUS_IDLE,
+                                        psutil.STATUS_SLEEPING):
+                    total_running_proc += 1
+            except psutil.NoSuchProcess:
+                logger.warn(f"(psutil) Process '{proc.name()}' exited unexpectedly.")

```

Comment:
Hi, I think we should also have generic exception handling in case to avoid failure.

Reply:
No such situation occurred so far. 
But this NoSuchProcess is identified during sspl_test restart sspl service and ERROR is logged. This is expected so we need to handle it.
In case of generic exception, it can occur anywhere in SSPL and its already handled in sspl_ll_d.

Reply:
@sumedhak27 Its a rare error in case a process gets exited while checking running process.
`psutil.process_iter()` is not robust against exiting process, so it needs to be in try/exception and our goal is to minimize the use of generic exceptions, that's why exception supplied from the library is used. 
Ref. 
https://github.com/giampaolo/psutil/issues/1811

Reply:
Hi, recently in iem case, we encountered the same issue, since we were handling only the specific exception, for an unhandled specific exception, which is caught in sspl_ll_d the whole sensor was turned off, so finally, Shriya added the generic expectation along with specific one, I wanted to mention such case @mariyappanp.

Reply:
Okay, Sumedh.